### PR TITLE
Browser runner (qbrt)

### PR DESCRIPTION
**Pull request by bgrins**
Just getting this up.  This implements the browser runner using a qbrt process as the platform.  There are still lots of papercuts here, similar to what we have with the dummy platform when loading in a browser tab.  Can you confirm this works for you after running:

```
npm install
npm run qbrt
```

By the way, I folded the startup into script into task-start instead of making a new one just to start this platform to limit code duplication.  If you like this, we might want to also fold the task-serve behavior into task-start with the platform flag, but I didn't add that to this PR.

### CODE REVIEW

File: build/tasks/task-start.js [hunk 1]
```diff
@@ -31,12 +31,19 @@ gulp.task('start', async () => {
...
+  } else {
+    const electron = process.platform === 'win32'
+      ? 'electron.cmd'
+      : 'electron';
+
+    return Promise.all([
+      spawn(electron, Paths.ELECTRON_RUNNER_DST_MAIN, runnerArgs, { logger }),
+      spawn('node', Paths.BROWSER_SERVER_DST_MAIN, args, { logger }),
+    ]);
+  }
```
Comment (victorporof) [inline build/tasks/task-start.js:48]: This is nice, let's fold the serve task in here too

File: package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
     "start": "cross-env NODE_ENV=production gulp build+start --obj-dir-name production-electron",
     "serve": "cross-env NODE_ENV=development gulp build+serve --obj-dir-name development-headless",
     "serve+logging": "cross-env NODE_ENV=development LOGGING=on gulp build+serve --obj-dir-name development-headless",
+    "qbrt": "cross-env NODE_ENV=production gulp build+start --platform qbrt --obj-dir-name production-electron",
+    "qbrt+logging": "cross-env NODE_ENV=production LOGGING=on gulp build+start --platform qbrt --obj-dir-name production-electron",
```
Comment (victorporof) [inline package.json]: These should be NODE_ENV=development, and have another start task that uses qbrt in production. But we're getting to a point where we have a ton of npm commands, and it's starting to get a bit hard to follow.

Maybe rename start to "start:electron" and have "start:qbrt" for production builds, while having "dev:electron" and "dev:qbrt" for development.

File: src/browser-runner/platform-qbrt/sagas/server-command-handlers.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import SharedActions from '../../../shared/actions/shared-actions';
+
+function create({ payload: { url } }) {
+  logger.log(`Chrome frontend hosted at ${url}`);
+  const qbrt = process.platform === 'win32'
+    ? 'qbrt.cmd'
+    : 'qbrt';
+  spawn(qbrt, 'run', [url], { logger });
+}
```
Comment (victorporof) [inline src/browser-runner/platform-qbrt/sagas/server-command-handlers.js]: There's a few more events that we should be handling here, but definitely followup material.

File: src/shared/widgets/web-contents/iframe-mozbrowser.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    super(...args);
+  }
+
+  componentDidMount() {
+    this._iframe.addEventListener('mozbrowserloadstart', () => {
+      this.props.onDidStartLoading();
+    });
+    this._iframe.addEventListener('mozbrowserloadend', () => {
+      this.props.onDidStopLoading();
+      this.props.onPageTitleSet({ title: 'Loaded' });
```
Comment (victorporof) [inline src/shared/widgets/web-contents/iframe-mozbrowser.jsx:28]: The reason we set the title to 'Loaded' in unprivileged web contexts is because of CORS limitations: we can't really get much information out from the iframe, and thus page titles are not accessible. I'd imagine in the case of qbrt and mozbrowser, we can do a lot better here?

**Review by victorporof — APPROVED** [review]
I'd be happy if we made some sense out of our growing npm commands list, but everything I've commented below can be followup material. Great work!

File: package.json (hunk 2 above)
Comment (victorporof) [reply package.json]: We could keep simple "start" and "dev" tasks too, those defaulting to electron.

File: src/shared/widgets/web-contents/iframe-mozbrowser.jsx (hunk 4 above)
Comment (victorporof) [reply src/shared/widgets/web-contents/iframe-mozbrowser.jsx:28]: Can we listen to DOMTitleSet events, and the whole plethora of mozbrowser events to minic the webview counterparts to an extent?

File: src/browser-runner/platform-qbrt/sagas/server-command-handlers.js (hunk 3 above)
Comment (victorporof) [reply src/browser-runner/platform-qbrt/sagas/server-command-handlers.js]: This also needs to send back a message to the server saying that a window has been created.

File: src/shared/widgets/web-contents/iframe-mozbrowser.jsx (hunk 4 above)
Comment (bgrins) [reply src/shared/widgets/web-contents/iframe-mozbrowser.jsx:28]: Those don't seem to be working in this version.  Might have something to do with it loading over http - need to spend some more time on it
